onnxscript/rewriter/generic_pattern.py

```diff
@@ -46,7 +42,7 @@ def __init__(
         self.pattern_outputs = pattern_outputs
         self.kwargs: dict[str, Any] = {}
 
-        matched_pattern_to_model_value: dict[ir.Value, ir.Value] = {}
+        matched_pattern_to_model_value: dict[str, ir.Value] = {}
```

Comment:
Is it possible when the name is None?

Reply:
Good question. For now, there is an assert. May be better to make it conditional update to map. Eventually, this should be replaced by the pattern IR (instead of onnx IR).

Reply:
In the end, we care about binding only users want, for which they will give a name